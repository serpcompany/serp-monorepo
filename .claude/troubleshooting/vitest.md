https://vitest.dev/guide/common-errors.html