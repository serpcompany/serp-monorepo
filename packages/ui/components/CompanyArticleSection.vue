<script setup lang="ts">
  defineProps<{
    article: string;
  }>();
</script>

<template>
  <UPageSection title="Buying Advice" class="bg-(--ui-bg)">
    <article v-if="article">
      <!-- eslint-disable vue/no-v-html -->
      <div
        role="document"
        class="prose dark:prose-invert mx-auto max-w-full"
        v-html="article"
      ></div>
    </article>
  </UPageSection>
</template>
