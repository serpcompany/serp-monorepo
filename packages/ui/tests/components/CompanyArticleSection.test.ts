import { describe, expect, it } from 'vitest'
import CompanyArticleSection from '../../components/CompanyArticleSection.vue'
import ComponentRender from '../componentRender'

describe('companyArticleSection Snapshot', () => {
  it.each([
    [
      'with valid article content',
      {
        props: {
          article:
            '<h1 id="article-title">Breaking News</h1><p>This is an amazing article content!</p>',
        },
      },
    ],
    ['with empty article', { props: { article: '' } }],
  ])(
    'renders %s correctly',
    async (desc: string, options: { props: unknown }) => {
      const html = await ComponentRender(
        `CompanyArticleSection ${desc}`,
        options,
        CompanyArticleSection,
      )
      expect(html).toMatchSnapshot()
    },
  )
})
