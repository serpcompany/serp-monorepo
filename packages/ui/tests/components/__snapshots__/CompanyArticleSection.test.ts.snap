// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CompanyArticleSection Snapshot > renders with empty article correctly 1`] = `
"<section data-orientation="vertical" class="relative isolate bg-(--ui-bg)">
  <div class="w-full max-w-(--ui-container) mx-auto px-4 sm:px-6 lg:px-8 flex flex-col lg:grid py-16 sm:py-24 lg:py-32 gap-8 sm:gap-16">
    <div class="">
      <!--v-if-->
      <!--v-if-->
      <h2 class="text-3xl sm:text-4xl lg:text-5xl text-pretty tracking-tight font-bold text-highlighted text-center">Buying Advice</h2>
      <!--v-if-->
      <!--v-if-->
      <!--v-if-->
    </div>
  </div>
</section>"
`;

exports[`CompanyArticleSection Snapshot > renders with valid article content correctly 1`] = `
"<section data-orientation="vertical" class="relative isolate bg-(--ui-bg)">
  <div class="w-full max-w-(--ui-container) mx-auto px-4 sm:px-6 lg:px-8 flex flex-col lg:grid py-16 sm:py-24 lg:py-32 gap-8 sm:gap-16">
    <div class="">
      <!--v-if-->
      <!--v-if-->
      <h2 class="text-3xl sm:text-4xl lg:text-5xl text-pretty tracking-tight font-bold text-highlighted text-center">Buying Advice</h2>
      <!--v-if-->
      <!--v-if-->
      <!--v-if-->
    </div>
    <article>
      <!-- eslint-disable vue/no-v-html -->
      <div role="document" class="prose dark:prose-invert mx-auto max-w-full">
        <h1 id="article-title">Breaking News</h1>
        <p>This is an amazing article content!</p>
      </div>
    </article>
  </div>
</section>"
`;
